### PR TITLE
Fix regression caused by #264

### PR DESCRIPTION
**Pull request by beeender**
The #264 caused some segment ratio tests fail. The entry's relevant
fields need to be set at the end of the iteration. Otherwise, only the
first seg will pass the condition check.

### CODE REVIEW

**Comment by beeender** [comment]
There is probably another issue, see https://dev2.ci.gpdb.pivotal.io/teams/gp-extensions/pipelines/dev.diskquota.mc_flaky_3.mulong/jobs/build_test_rhel6/builds/3

I have reverted the relevant changes and ran the test again. Let's see if they are regressions as well.

File: quotamodel.c [hunk 1]
```diff
@@ -982,56 +984,49 @@ calculate_table_disk_usage(bool is_init, HTAB *local_active_table_stat_map)
...
-				tsentry->owneroid = relowner;
+				transfer_table_for_quota(TableSizeEntryGetSize(tsentry, cur_segid), ROLE_QUOTA, (Oid[]){tsentry->owneroid},
+				                         (Oid[]){relowner}, cur_segid);
+				transfer_table_for_quota(TableSizeEntryGetSize(tsentry, cur_segid), ROLE_TABLESPACE_QUOTA,
+						(Oid[]){tsentry->owneroid, tsentry->tablespaceoid},
+						(Oid[]){relowner, reltablespace}, cur_segid);
 			}
-
-			if (tsentry->tablespaceoid != reltablespace)
+			if (cur_segid == (TableSizeEntrySegidEnd(tsentry) - 1))
```
Comment (beeender) [inline quotamodel.c:1026]: The tests failed. The original logic looks strange, would you please take a look? @fanfuxiaoran

File: quotamodel.c [hunk 2]
```diff
@@ -982,56 +984,49 @@ calculate_table_disk_usage(bool is_init, HTAB *local_active_table_stat_map)
...
+				transfer_table_for_quota(TableSizeEntryGetSize(tsentry, cur_segid), ROLE_QUOTA,
+				                         (Oid[]){tsentry->owneroid}, (Oid[]){relowner}, cur_segid);
+				transfer_table_for_quota(TableSizeEntryGetSize(tsentry, cur_segid), ROLE_TABLESPACE_QUOTA,
 				                         (Oid[]){tsentry->owneroid, tsentry->tablespaceoid},
-				                         (Oid[]){relowner, tsentry->tablespaceoid}, i);
-				tsentry->owneroid = relowner;
+				                         (Oid[]){relowner, reltablespace}, cur_segid);
 			}
-
-			if (tsentry->tablespaceoid != reltablespace)
```
Comment (fanfuxiaoran) [inline quotamodel.c:1027]: Why those codes are deleted? They process the tablespace quota things.

**Comment by beeender** [comment]
Replaced by #273

**Review by fanfuxiaoran — APPROVED** [review]

**Review by zhrt123 — APPROVED** [review]

**Pull request merged**